### ISSUE TITLE
Adapt file operations to accept `pathlib.Path` objects in addition to `str`s

### ISSUE THREAD

**Issue by erdmann** [libvips/pyvips#313]
The recommended way of manipulating paths these days in Python is with the standard library's [`pathlib`](https://docs.python.org/3/library/pathlib.html), and most of the rest of the standard library has been modified to accept `pathlib.Path` objects in addition to `str` when it comes to filenames.   `pyvips` doesn't like Path objects though:

```python
import pathlib
import pyvips
filename = pathlib.Path('example.tif')
im = pyvips.Image.new_from_file(filename)
# ---------------------------------------------------------------------------
# TypeError                                 Traceback (most recent call last)
# ...
# TypeError: initializer for ctype 'char *' must be a cdata pointer, not PosixPath

``` 

Consequently, I find my code littered with `pyvips.Image.something(str(filename))`.

I propose to modify pyvips so that it works with `pathlib.Path` objects in addition to the usual `str` where appropriate.  I'm happy to take this up and write a PR unless there's a reason not to.

What do you think?   Are there any gotchas you can think of?

**Comment by jcupitt** [comment]
Sure, that sounds like a useful modernisation.

It might be very simple: all strings passing from python into libvips go via this thing:

https://github.com/libvips/pyvips/blob/master/pyvips/error.py#L20-L30

That converts a python type into a utf-8 string. Maybe just adding a `str()` in there would do it? The case to test would be:

```python
x = pyvips.Image.jpegload(path)
```

ie. call a method which is generated at runtime.

Looking at `new_from_file` I think that would work with just the `to_bytes` change as well.

**Comment by erdmann** [comment]
Solved with PR #320.
